lib/library.cpp

```diff
@@ -703,7 +717,7 @@ Library::Error Library::loadFunction(const tinyxml2::XMLElement * const node, co
     return Error(OK);
 }
 
-bool Library::isargvalid(const Token *ftok, int argnr, const MathLib::bigint argvalue) const
+bool Library::isargvalid(const Token *ftok, int argnr, double argvalue) const
```

Comment:
I am skeptic about converting the integer to a double. It may work or not. I would suggest 2 isargvalid functions.

Reply:
I think the reason I didn't do this is because I added support for non-integer valid values in cfg files. So even if the argument to the function is an integer, the range it is compared to may or may not be an integer (at least in theory) which would complicate this. AFAiCS, there are no std-functions that use non-integer ranges on inputs so one solution is to remove support for that (essentially skip commit five in the PR). We would then know that the ranges are integers and the range checking would be much simpler. That would still male it possible to check if 1.00001 is passed to acos. Extending the valid range to non-integers could be added when there is need for it. Thoughts?

Reply:
> So even if the argument to the function is an integer, the range it is compared to may or may not be an integer (at least in theory) which would complicate this

true. I guess the `Library::isIntArgValid` function could check if the expression contains a dot somewhere and if there is, return the result of `Library::isFloatArgValid`.

Reply:
Done.